src/components/Posts/EditPostModal.tsx

```diff
@@ -0,0 +1,160 @@
+import {
+  Dialog,
+  DialogTitle,
+  Button,
+  DialogContent,
+  IconButton,
+  Box,
+  Stack,
+  TextField,
+  Typography,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import { useEffect, useState } from "react";
+import { useAppDispatch, useAppSelector } from "../../state/hooks";
+import { EmojiClickData } from "emoji-picker-react";
+import EmojiPickerIconButton from "../Common/EmojiPickerIconButton";
+import UserAvatar from "../Common/UserAvatar";
+import AddPhotoAlternateOutlinedIcon from "@mui/icons-material/AddPhotoAlternateOutlined";
+import axios from "axios";
+import { updatePost } from "../../state/slices/postsSlice";
+
+type EditPostModalProps = {
+  onClose: () => void;
+  open: boolean;
+  postId: number;
+};
+
+const styles = {
+  paperProps: {
+    overflow: "visible",
+    borderRadius: 20,
+  },
+  dialog: {
+    ".MuiDialog-scrollPaper": { alignItems: "flex-start" },
+  },
+  dialogTitle: {
+    display: "flex",
+    alignItems: "center",
+  },
+  titleText: {
+    fontWeight: "bold",
+    paddingLeft: 3,
+  },
+  dialogContentContainer: {
+    display: "flex",
+    paddingBottom: 2,
+    paddingLeft: 2,
+    paddingRight: 2,
+    paddingTop: 1,
+  },
+  avatar: { paddingRight: 1.5 },
+  textFieldContainer: { width: "100%", paddingTop: 0.5 },
+  textField: { paddingBottom: 2 },
+  postActions: {
+    display: "flex",
+    justifyContent: "space-between",
+    marginTop: 1,
+  },
+};
+
+const EditPostModal = ({ onClose, open, postId }: EditPostModalProps) => {
+  const user = useAppSelector((state) => state.user);
+  const dispatch = useAppDispatch();
+  const existingPost = useAppSelector((state) =>
+    state.posts.posts.find((post) => post.postId === postId)
+  );
+  const [postTextContent, setPostTextContent] = useState<string>("");
```

Comment:
This is just so you know, but you need to put the non-null assertion at the end for this to work. Adding the useEffect below adds more code for no reason. Also, we probably don't need to type the useState here:

```suggestion
  const [postTextContent, setPostTextContent] = useState(existingPost.textContent!);
```

If we go with my suggestion above to apply the non-null assertion on the `exisingPost` object itself we won't need to do any of this anyways

Reply:
Done